packages/require/lib/compiler.ts

```diff
@@ -0,0 +1,94 @@
+import path from "path";
+import TruffleError from "@truffle/error";
+
+import type { CreateOptions, Service } from "ts-node";
+import TruffleConfig from "@truffle/config";
+import { readFileSync, existsSync } from "fs";
+
+import originalRequire from "original-require";
+
+import vm from "vm";
+
+/**
+ * @returns compiler function from user's installation of `ts-node` if script is
+ * a ts or tsx file, otherwise `undefined`.
+ */
+export function compile(conf: TruffleConfig, scriptPath: string): string {
+  const source = readFileSync(scriptPath, { encoding: "utf-8" });
+
+  // we only compile TS files, so just return the source unless we are dealing
+  // with one of those
+  if (![".ts", ".tsx", ".cts"].includes(path.extname(scriptPath))) {
```

Comment:
By "a constant," I assume you mean a constant variable rather than a literal?

I tend to follow the rule of three for things like this, but it's easy enough to move it for future use.